pkg/resiliency/retry.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resiliency
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/dapr/kit/retry"
+)
+
+type CodeError struct {
+	StatusCode int32
+	err        error
+}
+
+func (ce CodeError) Error() string {
+	return ce.err.Error()
+}
+
+func (ce CodeError) Unwrap() error {
+	return ce.err
+}
+
+func NewCodeError(statusCode int32, err error) CodeError {
+	return CodeError{
+		StatusCode: statusCode,
+		err:        err,
+	}
+}
+
+type Retry struct {
+	*retry.Config
+	*StatusCodeFilter
+}
+
+func NewRetry(retryConfig *retry.Config, statusCodeFilter *StatusCodeFilter) *Retry {
+	return &Retry{
+		Config:           retryConfig,
+		StatusCodeFilter: statusCodeFilter,
+	}
+}
+
+type StatusCodeFilter struct {
+	retryOnPatterns  []*regexp.Regexp
```

Comment:
Make these not pointers: `[]regexp.Regexp`

Reply:
done.